feat: add experimental otlp metrics exporter

xuan-cao-swi opened the pull request:
Follow-up PR for https://github.com/open-telemetry/opentelemetry-ruby/pull/1545 (move experimental feature into separate gem)

## Description

Added otlp metrics exporter that works with otlp collector through http protocol.

Most code copied from (trace) exporter (with test as well) with the assumption of same ssl, http etc. configuration. The major change is the proto decode. If the log exporter also use the similar configuration, having a utility file can help DRY.

Quick demo
```ruby
require 'opentelemetry/sdk'
require 'opentelemetry-metrics-sdk' # need to install this from building the metrics_sdk and metrics_api folder
require 'opentelemetry/exporter/otlp'

OpenTelemetry::SDK.configure

otlp_metric_exporter = OpenTelemetry::Exporter::OTLP::MetricsExporter.new

OpenTelemetry.meter_provider.add_metric_reader(otlp_metric_exporter)

meter = OpenTelemetry.meter_provider.meter("SAMPLE_METER_NAME")

histogram = meter.create_histogram('histogram', unit: 'smidgen', description: 'desscription')

histogram.record(123, attributes: {'foo' => 'bar'})

OpenTelemetry.meter_provider.metric_readers.each(&:pull)
OpenTelemetry.meter_provider.shutdown
```

Testing failed but it will success once the https://github.com/open-telemetry/opentelemetry-ruby/pull/1532 is merged.

kaylareopelle commented:
The jaeger CI failure is now fixed on `main`.

kaylareopelle reviewed (changes requested):
This is awesome, Xuan! The exporter worked great with a local OTLP collector. 

Initially, I had some issues with New Relic's OTLP endpoint. Adding the `aggregation_temporality` attribute to the protobuf payload resolved the problem. I have a few suggestions below that add `aggregation_temporality` to the exporter. They're accompanied by #1554. That PR adds `aggregation_temporality` to `MetricData` so that it can be accessed using `metrics.aggregation_temporality`.

kaylareopelle commented on exporter/otlp-metrics/CHANGELOG.md:
```diff
@@ -0,0 +1,5 @@
+# Release History: opentelemetry-exporter-otlp-metrics
+
+### v0.0.1 / 2023-12-07
+
+* Initial release.
```
It follows what was done on the [sequel instrumentation initial changelog](https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/672/files#diff-3aef9e0af2cba0cb58d065d61f188ab3487a254ca8915ced27609b4e9cb748a0)

xuan-cao-swi commented on exporter/otlp-metrics/test/opentelemetry/exporter/otlp/metrics_exporter_test.rb:
```diff
@@ -0,0 +1,626 @@
+# frozen_string_literal: true
```
Sure

xuan-cao-swi commented:
@kaylareopelle Thanks for the review, I have added the change based on [aggregation_temporality](https://github.com/open-telemetry/opentelemetry-ruby/pull/1554), also added the gem test to ci.yml.

xuan-cao-swi commented on exporter/otlp-metrics/opentelemetry-exporter-otlp-metrics.gemspec:
```diff
@@ -0,0 +1,56 @@
+  spec.add_dependency 'google-protobuf', '~> 3.14'
+  spec.add_dependency 'opentelemetry-api', '~> 1.1'
+  spec.add_dependency 'opentelemetry-common', '~> 0.20'
+  spec.add_dependency 'opentelemetry-metrics-api'
+  spec.add_dependency 'opentelemetry-metrics-sdk'
```
Probably should remove these dependencies? 

Since there is no metrics-related gem in rubygems, user who install this and use it will fail.

User who want to use it should create the metrics gem by themselves, and require metrics gem first, and then this metrics exporter.

mwear reviewed (approved):
If there are no objections, I'll merge this at EOD. Thanks @xuan-cao-swi.
